maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -1204,40 +1211,57 @@ void toolchains(CliRequest cliRequest) throws Exception {
                         "The specified user toolchains file does not exist: " + userToolchainsFile);
             }
         } else {
-            userToolchainsFile = DEFAULT_USER_TOOLCHAINS_FILE;
+            String userToolchainsFileStr = cliRequest.getUserProperties().getProperty(Constants.MAVEN_USER_TOOLCHAINS);
+            if (userToolchainsFileStr != null) {
+                userToolchainsFile = new File(userToolchainsFileStr);
+            }
         }
 
-        File globalToolchainsFile;
+        File installationToolchainsFile = null;
```

Comment:
Why not use `Path` here throughout?

Reply:
No reason, but I'd like to not expand the scope of this PR too much.  I'd rather raise another PR for doing a full pass on `MavenCli` to switch to nio2 Path.

Reply:
Agreed.